aidial_adapter_bedrock/llm/model/adapter.py

```diff
@@ -106,23 +111,31 @@ async def get_bedrock_adapter(
                 await Bedrock.acreate(aws_client_config), model
             )
         case (
-            ChatCompletionDeployment.META_LLAMA2_13B_CHAT_V1
-            | ChatCompletionDeployment.META_LLAMA2_70B_CHAT_V1
+            ChatCompletionDeployment.META_LLAMA3_8B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_70B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_1_8B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_2_1B_INSTRUCT_V1
+            | ChatCompletionDeployment.META_LLAMA3_2_3B_INSTRUCT_V1
         ):
-            return MetaAdapter.create(
-                await Bedrock.acreate(aws_client_config), model, llama2_config
+            return ConverseAdapter(
+                deployment=model,
+                bedrock=await Bedrock.acreate(aws_client_config),
+                storage=create_file_storage(api_key),
+                input_tokenizer_factory=llama_tokenizer_factory,
+                support_tools=False,
```

Comment:
Ah, well, it's very annoying. 

Llama models are very customizable - you are supposed to configure system prompt for tool calling yourself: https://www.llama.com/docs/model-cards-and-prompt-formats/llama3_2

And the light-weight models are supposed to support tool calling.
So it's probable that Bedrock messed up the system prompt on their side in Converse-to-Llama adapter.

Once again when we switch to Converse API, we lose a lot of control - it's now impossible to fully control the system prompt, not possible to count tokens on our side, impossible to troubleshoot...

Reply:
That's valid point

If that's really a miss from their side, I will be forced to move this models back to `MetaAdapter` with vanilla `invokeModel`

Created issue for that:
https://github.com/epam/ai-dial-adapter-bedrock/issues/181